### PR TITLE
Draft: feat: Add support for reading row is hidden

### PR DESCRIPTION
**Pull request by macsj200**
This change adds support to optionally read if a row is hidden or not.

### CODE REVIEW

File: src/main/scala/com/crealytics/spark/excel/v2/ExcelHelper.scala [hunk 1]
```diff
@@ -216,6 +216,10 @@ class ExcelHelper private (options: ExcelOptions) {
       if (options.columnNameOfRowNumber.isDefined) Vector[String](options.columnNameOfRowNumber.get)
       else Vector.empty[String]
 
+    val rowIsHiddenColumn =
+      if (options.columnNameOfRowIsHidden.isDefined) Vector[String](options.columnNameOfRowIsHidden.get)
+      else Vector.empty[String]
```
Comment (nightscape) [inline src/main/scala/com/crealytics/spark/excel/v2/ExcelHelper.scala:221]:
```suggestion
    val rowIsHiddenColumn = options.columnNameOfRowIsHidden.toSeq
```

File: src/main/scala/com/crealytics/spark/excel/v2/ExcelInferSchema.scala [hunk 2]
```diff
@@ -39,8 +39,17 @@ class ExcelInferSchema(val options: ExcelOptions) extends Serializable {
         Vector[StructField](StructField(options.columnNameOfRowNumber.get, IntegerType, false))
       } else Vector.empty
 
+    /* Possible StructField for row-is-hidden column */
+    val rowIsHiddenField =
+      if (options.columnNameOfRowIsHidden.isDefined) {
+        Vector[StructField](StructField(options.columnNameOfRowIsHidden.get, BooleanType, false))
+      } else Vector.empty
```
Comment (nightscape) [inline src/main/scala/com/crealytics/spark/excel/v2/ExcelInferSchema.scala:46]:
```suggestion
    val rowIsHiddenField = options.columnNameOfRowIsHidden.map(cn => StructField(cn, BooleanType, false)).toSeq
```

File: src/main/scala/com/crealytics/spark/excel/v2/ExcelInferSchema.scala [hunk 3]
```diff
@@ -39,8 +39,17 @@ class ExcelInferSchema(val options: ExcelOptions) extends Serializable {
...
+
+    /* Header without row-is-hidden-column */
+    var dataHeader = if (options.columnNameOfRowIsHidden.isDefined) header.tail else header
+
```
Comment (nightscape) [inline src/main/scala/com/crealytics/spark/excel/v2/ExcelInferSchema.scala:50]:
```suggestion

    /* Header without row-is-hidden-column and row-number-column */
    val dataHeader = header.drop(options.columnNameOfRowIsHidden.size + options.columnNameOfRowNumber.size)
```

File: src/main/scala/com/crealytics/spark/excel/v2/ExcelHelper.scala (hunk 1 above)
Comment (macsj200) [reply src/main/scala/com/crealytics/spark/excel/v2/ExcelHelper.scala]: I incorporated this suggestion. It caused a type error with the returned value below, but I changed it to concatenate `rowIsHiddenColumn.toVector` and it seems to compile now. I'm not sure what the difference is between Sequence and Vector, but it seems simple enough to convert from one to the other.

**Comment by macsj200** [comment]
Thanks for the suggestions! I made some changes, and I'm running the test suite locally now.

**Comment by nightscape** [comment]
@macsj200 there's some more room for improvement w.r.t. replacing `var`s by `val`s and `Option` handling.
Do you want to use this opportunity to flex your Scala muscle? 😃

**Comment by macsj200** [comment]
Thanks again for the review. Unfortunately I don't have much scala muscle to speak of, let alone flex. I don't think this is the right time for me to work on this, perhaps I can take it up at a later point.

**Pull request closed without merging**